File: java/src/Ice/src/main/java/IceSSL/Instance.java
// **********************************************************************
//
// Copyright (c) 2003-2018 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package IceSSL;

class Instance extends IceInternal.ProtocolInstance
{
    Instance(SSLEngine engine, short type, String protocol)
    {
        super(engine.communicator(), type, protocol, true);
        _engine = engine;
    }

    SSLEngine engine()
    {
        return _engine;
    }

    int securityTraceLevel()
    {
        return _engine.securityTraceLevel();
    }

    String securityTraceCategory()
    {
        return _engine.securityTraceCategory();
    }

    boolean initialized()
    {
        return _engine.initialized();
    }

    javax.net.ssl.SSLEngine createSSLEngine(boolean incoming, java.net.InetSocketAddress peerAddr)
    {
        return _engine.createSSLEngine(incoming, peerAddr);
    }

    void traceConnection(java.nio.channels.SocketChannel fd, javax.net.ssl.SSLEngine engine, boolean incoming)
    {
        _engine.traceConnection(fd, engine, incoming);
    }

    void verifyPeer(NativeConnectionInfo info, java.nio.channels.SelectableChannel fd, String address)
    {
        _engine.verifyPeer(info, fd, address);
    }

    void trustManagerFailure(boolean incoming, java.security.cert.CertificateException ex)
        throws java.security.cert.CertificateException
    {
        _engine.trustManagerFailure(incoming, ex);
    }

    private SSLEngine _engine;
}

File: csharp/src/Ice/IceSSL/Instance.cs
// Copyright (c) ZeroC, Inc.

using System.Net.Security;
using System.Security.Cryptography.X509Certificates;

namespace IceSSL;

internal class Instance : IceInternal.ProtocolInstance
{
    internal Instance(SSLEngine engine, short type, string protocol) :
        base(engine.communicator(), type, protocol, true) => _engine = engine;

    internal SSLEngine engine() => _engine;

    internal int securityTraceLevel() => _engine.securityTraceLevel();

    internal string securityTraceCategory() => _engine.securityTraceCategory();

    internal bool initialized() => _engine.initialized();

    internal X509Certificate2Collection certs() => _engine.certs();

    internal int checkCRL() => _engine.checkCRL();

    internal void traceStream(SslStream stream, string connInfo) => _engine.traceStream(stream, connInfo);

    internal void verifyPeer(ConnectionInfo info, string description) =>
        _engine.verifyPeer(info, description);

    internal Ice.InitializationData initializationData() =>
        IceInternal.Util.getInstance(_engine.communicator()).initializationData();

    private readonly SSLEngine _engine;
}
